File: tests/testdata/codegen/assets-archives-pp/python/assets-archives.py
import khulnasoft
import khulnasoft_aws as aws

site_bucket = aws.s3.Bucket("siteBucket")
test_file_asset = aws.s3.BucketObject("testFileAsset",
    bucket=site_bucket.id,
    source=khulnasoft.FileAsset("file.txt"))
test_string_asset = aws.s3.BucketObject("testStringAsset",
    bucket=site_bucket.id,
    source=khulnasoft.StringAsset("<h1>File contents</h1>"))
test_remote_asset = aws.s3.BucketObject("testRemoteAsset",
    bucket=site_bucket.id,
    source=khulnasoft.RemoteAsset("https://khulnasoft.test"))
test_file_archive = aws.lambda_.Function("testFileArchive",
    role=site_bucket.arn,
    code=khulnasoft.FileArchive("file.tar.gz"))
test_remote_archive = aws.lambda_.Function("testRemoteArchive",
    role=site_bucket.arn,
    code=khulnasoft.RemoteArchive("https://khulnasoft.test/foo.tar.gz"))
test_asset_archive = aws.lambda_.Function("testAssetArchive",
    role=site_bucket.arn,
    code=khulnasoft.AssetArchive({
        "file.txt": khulnasoft.FileAsset("file.txt"),
        "string.txt": khulnasoft.StringAsset("<h1>File contents</h1>"),
        "remote.txt": khulnasoft.RemoteAsset("https://khulnasoft.test"),
        "file.tar": khulnasoft.FileArchive("file.tar.gz"),
        "remote.tar": khulnasoft.RemoteArchive("https://khulnasoft.test/foo.tar.gz"),
        ".nestedDir": khulnasoft.AssetArchive({
            "file.txt": khulnasoft.FileAsset("file.txt"),
            "string.txt": khulnasoft.StringAsset("<h1>File contents</h1>"),
            "remote.txt": khulnasoft.RemoteAsset("https://khulnasoft.test"),
            "file.tar": khulnasoft.FileArchive("file.tar.gz"),
            "remote.tar": khulnasoft.RemoteArchive("https://khulnasoft.test/foo.tar.gz"),
        }),
    }))

File: tests/testdata/codegen/assets-archives-pp/nodejs/assets-archives.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as aws from "@khulnasoft/aws";

const siteBucket = new aws.s3.Bucket("siteBucket", {});
const testFileAsset = new aws.s3.BucketObject("testFileAsset", {
    bucket: siteBucket.id,
    source: new khulnasoft.asset.FileAsset("file.txt"),
});
const testStringAsset = new aws.s3.BucketObject("testStringAsset", {
    bucket: siteBucket.id,
    source: new khulnasoft.asset.StringAsset("<h1>File contents</h1>"),
});
const testRemoteAsset = new aws.s3.BucketObject("testRemoteAsset", {
    bucket: siteBucket.id,
    source: new khulnasoft.asset.RemoteAsset("https://khulnasoft.test"),
});
const testFileArchive = new aws.lambda.Function("testFileArchive", {
    role: siteBucket.arn,
    code: new khulnasoft.asset.FileArchive("file.tar.gz"),
});
const testRemoteArchive = new aws.lambda.Function("testRemoteArchive", {
    role: siteBucket.arn,
    code: new khulnasoft.asset.RemoteArchive("https://khulnasoft.test/foo.tar.gz"),
});
const testAssetArchive = new aws.lambda.Function("testAssetArchive", {
    role: siteBucket.arn,
    code: new khulnasoft.asset.AssetArchive({
        "file.txt": new khulnasoft.asset.FileAsset("file.txt"),
        "string.txt": new khulnasoft.asset.StringAsset("<h1>File contents</h1>"),
        "remote.txt": new khulnasoft.asset.RemoteAsset("https://khulnasoft.test"),
        "file.tar": new khulnasoft.asset.FileArchive("file.tar.gz"),
        "remote.tar": new khulnasoft.asset.RemoteArchive("https://khulnasoft.test/foo.tar.gz"),
        ".nestedDir": new khulnasoft.asset.AssetArchive({
            "file.txt": new khulnasoft.asset.FileAsset("file.txt"),
            "string.txt": new khulnasoft.asset.StringAsset("<h1>File contents</h1>"),
            "remote.txt": new khulnasoft.asset.RemoteAsset("https://khulnasoft.test"),
            "file.tar": new khulnasoft.asset.FileArchive("file.tar.gz"),
            "remote.tar": new khulnasoft.asset.RemoteArchive("https://khulnasoft.test/foo.tar.gz"),
        }),
    }),
});
